src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/HPackEncoder.cs

```diff
@@ -2,12 +2,168 @@
 // Licensed under the Apache License, Version 2.0.
 // See THIRD-PARTY-NOTICES.TXT in the project root for license information.
 
+using System.Collections.Generic;
 using System.Diagnostics;
 
 namespace System.Net.Http.HPack
 {
-    internal static class HPackEncoder
+    internal class HPackEncoder
     {
+        private IEnumerator<KeyValuePair<string, string>> _enumerator;
```

Comment:
Is there any way we can get rid of this allocation? I know we haven't optimized the HTTP/2 performance as yet but I'm sure this will come up.

Reply:
This is directly ported from kestrel. We can create a follow-up issue/ pull request to remove this allocation. I'd rather not do it in this PR as I already have done too much.